msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,188 @@
+import org.openscience.cdk.interfaces.IMolecularFormula;
+import org.openscience.cdk.silent.SilentChemObjectBuilder;
+import org.openscience.cdk.tools.manipulator.MolecularFormulaManipulator;
+
+public class SiriusIdentificationMethod implements MSDKMethod<List<IonAnnotation>> {
+
+  private final Sirius sirius;
+  private MsSpectrum ms1;
+  private MsSpectrum ms2;
+  private double parentMass;
+  private String ion;
+  private int numberOfCandidates;
+  private List<IonAnnotation> result;
+
+  public SiriusIdentificationMethod(@Nullable MsSpectrum ms1, MsSpectrum ms2, double parentMass,
+      String ion) {
+    sirius = new Sirius();
+    this.ms1 = ms1;
+    this.ms2 = ms2;
+    this.parentMass = parentMass;
+    this.ion = ion;
+    numberOfCandidates = 5;
+  }
+
+  public double getParentMass() {
+    return parentMass;
+  }
+
+  public void setParentMass(double mass) {
+    parentMass = mass;
+  }
+
+  public String getIonization() {
+    return ion;
+  }
+
+  public void setIonization(String ion) {
+    this.ion = ion;
+  }
+
+  public MsSpectrum getMsSpectrum() {
+    return ms1;
+  }
+
+  public void setMsSpectrum(MsSpectrum ms1) {
+    this.ms1 = ms1;
+  }
+
+  public MsSpectrum getMs2Spectrum() {
+    return ms2;
+  }
+
+  public void setMs2Spectrum(MsSpectrum ms2) {
+    this.ms2 = ms2;
+  }
+
+  public int getNumberOfCandidates() {
+    return numberOfCandidates;
+  }
+
+  public void setNumberOfCandidates(int number) {
+    numberOfCandidates = number;
+  }
+
+  /**
+   * This function is left here for custom spectrum files (just columns of mz and intensity values)
+   * Does similar to mgf parser functionality
+   */
+  public static MsSpectrum readCustomMsFile(File file, String delimeter)
+      throws IOException, MSDKRuntimeException {
+    Scanner sc = new Scanner(file);
+    ArrayList<String> strings = new ArrayList<>();
+    while (sc.hasNext()) {
+      strings.add(sc.nextLine());
+    }
+    sc.close();
+
+    int size = strings.size();
+    double mz[] = new double[size];
+    float intensity[] = new float[size];
+
+    int index = 0;
+    for (String s : strings) {
+      String[] splitted = s.split(delimeter);
+      if (splitted.length == 2) {
+        mz[index] = Double.parseDouble(splitted[0]);
+        intensity[index++] = Float.parseFloat(splitted[1]);
+      } else {
+        throw new MSDKRuntimeException("Incorrect spectrum structure");
+      }
+    }
+
+    MsSpectrumType type = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mz, intensity, size);
+    return new SimpleMsSpectrum(mz, intensity, size, type);
+  }
+
+
+  /**
+   * Transformation of MSDK data structures into Sirius structures and processing by sirius
+   */
+  protected List<IdentificationResult> siriusProcessSpectrums() throws MSDKException {
+    Spectrum<Peak> siriusMs1 = null, siriusMs2;
+    siriusMs2 = sirius
+        .wrapSpectrum(ms2.getMzValues(), LocalArrayUtil.convertToDoubles(ms2.getIntensityValues()));
```

Comment:
It's a bad practice to have many function calls on a single line, because when an exception occurs on that line, you won't know where it came from. It's better to split this into several lines (first get ms2.getMzValues(), then get the intensity values, then call sirius.wrapSpectrum...)

Reply:
An alternative solution is for `sirius.wrapSpectrum` to take just `ms2`, and then call `ms2.getMzValues()` and `LocalArrayUtil.convertToDoubles(ms2.getIntensityValues())`.

Reply:
We cannot change sirius, is's coming from an imported library.